integrations/api_repo_test.go

```diff
@@ -46,6 +46,220 @@ func TestAPISearchRepoNotLogin(t *testing.T) {
 		assert.Contains(t, repo.Name, keyword)
 		assert.False(t, repo.Private)
 	}
+
+	// Should return all (max 50) public repositories
+	req = NewRequest(t, "GET", "/api/v1/repos/search?limit=50")
```

Comment:
This is a lot of duplicated code. Maybe we should add a helper function
```
func APIRepoSearchResult(t testing.T, params string, expectedLen int) *api.SearchResults
...
RepoSearchResult(t, "?limit=50", 12)
```

Even better, add a session parameter, so you can use it for the logged-in tests too.

Reply:
@ethantkoenig If you want that you can review https://github.com/go-gitea/gitea/pull/2575 as it rewrites tests. I wanted to change as few as possible to make reviewing easier, but if it is desired the mentioned PR does this ;-)